storage/2023-11-03/blob/containers/create.go

```diff
@@ -41,6 +44,30 @@ func (c Client) Create(ctx context.Context, containerName string, input CreateIn
 		return
 	}
 
+	// Retry the container creation if a conflicting container is still in the process of being deleted
+	retryFunc := func(resp *http.Response, _ *odata.OData) (bool, error) {
+		if resp != nil {
+			if resp.StatusCode == http.StatusConflict {
+				// TODO: move this error response parsing to a common helper function
+				respBody, err := io.ReadAll(resp.Body)
+				if err != nil {
+					return false, fmt.Errorf("could not parse response body")
+				}
+				resp.Body.Close()
+				respBody = bytes.TrimPrefix(respBody, []byte("\xef\xbb\xbf"))
+				res := ErrorResponse{}
+				if err = xml.Unmarshal(respBody, &res); err != nil {
+					return false, err
+				}
+				resp.Body = io.NopCloser(bytes.NewBuffer(respBody))
+				if res.Code != nil {
+					return strings.Contains(*res.Code, "ContainerBeingDeleted"), nil
+				}
```

Comment:
Note: it might be worth changing RetryFunc so that it's passed a `client.Response` rather than an `http.Response`, which would make common parsing a little cleaner. I can do that as part of this change if anyone thinks that's warranted?